torch/csrc/autograd/custom_function.cpp

```diff
@@ -0,0 +1,84 @@
+#include <torch/csrc/autograd/custom_function.h>
+#include <torch/csrc/autograd/functions/accumulate_grad.h>
+
+namespace torch { namespace autograd {
+
+std::vector<Variable> _wrap_outputs(const std::unordered_set<at::TensorImpl*> &inputs,
+  const std::unordered_set<at::TensorImpl*> &non_differentiable,
+  const std::unordered_set<at::TensorImpl*> &dirty_inputs,
+  const std::vector<Variable> &raw_outputs,
+  std::shared_ptr<Function> cdata) {
```

Comment:
C++ nit: you can avoid the refcount bump by passing this shared pointer by const reference. I also don't think it matters here, but it's a pretty common idiom in C++ and worth knowing about (since lots of refcount bumps add up!)